spring-cloud-release-tools-core/src/main/java/org/springframework/cloud/release/cloud/docs/SpringCloudCustomProjectDocumentationUpdater.java

```diff
@@ -67,7 +67,7 @@ public boolean isApplicable(File clonedDocumentationProject,
 	 * used
 	 */
 	@Override
-	public File updateDocsRepo(File clonedDocumentationProject,
+	public File updateDocsRepoForReleaseTrain(File clonedDocumentationProject,
```

Comment:
Cause it's a file that i operate on. Since most of those folders are using git underneath and git abstractions are file based afair I'm also reusing it. I don't see any reason to use resource here.

Reply:
But this is a good example of where you then tie releaser to the Spring Cloud process by tying your abstractions to Git for the docs update.  For the abstraction, I'd recommend moving to a `Resource` to allow for extensibility.

Reply:
However crazy it sounds, I'm not a fan of using Spring all over the place :P You can see that the core of the releaser uses no Spring at all. It's the `-spring` module that binds it all together in configurations and sets up beans. But that's just me.

Reply:
This isn't about Spring using it all over the place.  It's about defining good, versatile abstractions.  By tying this (and anywhere else you do this) to a `File` limits the ability to extend it.

Reply:
There was no point to extend it whatsover, it's just a file on the disk. However, I can change it to `Resource` if you think that it makes more sense.

Reply:
To be clear, it's not this impl that I'm making a big deal about...it's the abstraction you are extending from (since this is overrding a method somewhere else).